bench/cljam/io/bam/decoder_bench.clj

```diff
@@ -0,0 +1,21 @@
+(ns cljam.io.bam.decoder-bench
+  (:require [libra.bench :refer [defbench are]]
+            [libra.criterium :as c]
+            [cljam.test-common :as tcommon]
+            [cljam.io.sam :as sam]))
+
+(defbench decode-bam-alignment-short-bench
+  (are [f]
+       (c/quick-bench
+        (with-open [r (sam/reader f)]
+          (dorun (sam/read-alignments r))))
+    tcommon/test-bam-file
+    tcommon/medium-bam-file
+    tcommon/codec-benchmark-bam-file))
+
+(defbench decode-bam-alignment-long-bench
+  (are [f]
+       (c/quick-bench
+        (with-open [r (sam/reader f)]
+          (dorun (vec (take 10000000 (sam/read-alignments r))))))
+    tcommon/large-bam-file))
```

Comment:
You need to download cavia-managed files first

Reply:
I'm assuming `cavia/get!` is called from these `tcommon/prepare-cavia!` and the file will be downloaded.
https://github.com/chrovis/cljam/blob/157df000bb972790477793ca1dd23b6dacdcff46/bench/cljam/io/sam_bench.clj#L24-L25

https://github.com/chrovis/cljam/blob/157df000bb972790477793ca1dd23b6dacdcff46/bench/cljam/io/sam_bench.clj#L51-L52

I coded the following functions as a reference.
https://github.com/chrovis/cljam/blob/843e30b4aaf34e2fe79d070f1a2122097eb78b99/bench/cljam/algo/pileup_bench.clj#L9-L10

if my understanding is mistaken, please tell me. 🙏

Reply:
My bad, I missed this commit 🙇  
https://github.com/chrovis/cljam/compare/912535f3ab363b260c3e99ff329dc6fefacf9bc6..c80d76e5e431e3a2d54466024a26e5cf1c6c0050
BTW `with-before-after` is doing nothing in this case actually, so just calling `(tcommon/prepare-cavia!)` will suffice.